src/_P016_IR.ino

```diff
@@ -137,16 +137,26 @@ boolean Plugin_016(byte function, struct EventStruct *event, String& string)
           irReceiver->setUnknownThreshold(kMinUnknownSize); // Ignore messages with less than minimum on or off pulses.
           irReceiver->enableIRIn(); // Start the receiver
         }
-        if (irReceiver != 0 && irPin == -1)
+        if (irReceiver != 0 && irPin == -1) // what is the purpose of this?
```

Comment:
You need to check a pointer before de-referencing it, so it cannot be 0.
A pin which is not set, is defined as -1.
So if there is no pin set and the pointer object exists, we should delete the object